### PR TITLE
[winlogbeat] Retry EvtSubscribe from start if fails with strict mode

### PR DESCRIPTION
**Pull request by marc-gr**
## What does this PR do?

Creates a subscription to the winlog channel using `EvtSubscribeStrict` flag to be able to notice when a bookmark ran invalid. In that scenario we retry the subscription from the beginning of the channel.

## Why is it important?

Based on https://github.com/elastic/beats/issues/29793 there is the suspicion that under some scenarios winlogbeat might be ignoring some events when an invalid bookmark is used.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.
## Related issues

- Closes https://github.com/elastic/beats/issues/29793

### CODE REVIEW

**Comment by marc-gr** [comment]
@andrewkroh the retry is performed automatically in this first approach. Do you think adding a way to enable or disable this behavior at will would be preferred?

**Comment by andrewkroh** [comment]
That looks pretty simple so perhaps it does not need to be configurable. Can you create an integration test with windows that exercises this path. One idea I have is to read from a test log we create during the tests, modify the bookmark we have to move the read index forward such that it's become an invalid bookmark, then try to read using the bookmark, verify it starts at the beginning.

File: winlogbeat/eventlog/wineventlog.go [hunk 1]
```diff
@@ -215,6 +217,12 @@ func (l *winEventLog) openChannel(bookmark win.EvtHandle) error {
 		l.query,  // Query - nil means all events
 		bookmark, // Bookmark - for resuming from a specific event
 		flags)
+
+	if err == win.ERROR_NOT_FOUND {
+		// The bookmarked event was not found, we retry the subscription from the start.
```
[the remark this answers is not in the document]
Comment (marc-gr) [reply winlogbeat/eventlog/wineventlog.go:222]: 👍 Will probably add a warning log also

**Comment by andrewkroh** [comment]
Based on https://github.com/elastic/beats/issues/30201 I think another test case for this should be reading from a channel, deleting the channel, then recreating the channel and read from it again with the previously saved bookmark. I think that's a realistic scenario that is happening to users based on that report.

**Comment by nicpenning** [comment]
Let me know if I should create a new issue but have you ever noticed that Winlogbeat can't read from the SysMon channel when SysMon is recently installed or when the configuration changes? A restart of the winlogbeat service is needed when making changes to SysMon. It's been many versions since I tested this but I will report back if that's still happening. The only way we could successfully deploy 9k+ winlogbeat agents and SysMon was to install sysmon first then winlogbeat. Then any changes to SysMon would require us to restart winlogbeat so we can gather events as soon as possible.

File: winlogbeat/beater/winlogbeat.go [hunk 2]
```diff
@@ -98,7 +98,7 @@ func (eb *Winlogbeat) init(b *beat.Beat) error {
 		if err != nil {
 			return fmt.Errorf("failed to create new event log: %w", err)
 		}
-		eb.log.Debugf("Initialized EventLog]", eventLog.Name())
+		eb.log.Debugf("Initialized EventLog %s", eventLog.Name())
```
Comment (andrewkroh) [inline winlogbeat/beater/winlogbeat.go:101]: How about this in following along with the related changes that broke this (https://github.com/elastic/beats/commit/3d66cc1074a91262704dd739d388e67515e3992d#diff-420d1f0b3025a4724f79acfc014595527b87b1e352cd30dce6dce918d25af023R75).

```suggestion
		eb.log.Debugw("Initialized EventLog", "id", eventLog.Name())
```

File: winlogbeat/tests/system/test_wineventlog.py [hunk 3]
```diff
@@ -63,6 +63,91 @@ def test_resume_reading_events(self):
...
+        event_logs = self.read_registry(requireBookmark=True)
+        self.assertTrue(len(list(event_logs.keys())), 1)
+        self.assertIn(self.providerName, event_logs)
+        record_number = event_logs[self.providerName]["record_number"]
+        self.assertTrue(record_number, 2)
+
+        msg3 = "Third event"
+        self.write_event_log(msg3)
+
+        evts = self.read_events()
```
[the remark this answers is not in the document]
Comment (marc-gr) [reply winlogbeat/tests/system/test_wineventlog.py:99]: The triggered error is the same in the case where the channel is recreated and the bookmark is invalid, I was not able to re-create a scenario to trigger the other documented ones, which I added for completeness

**Review by andrewkroh — APPROVED** [review]

**Pull request merged**